internal/pkg/config/environment_test.go

```diff
@@ -0,0 +1,96 @@
+/*******************************************************************************
+ * Copyright 2019 Dell Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *******************************************************************************/
+
+package config
+
+import (
+	"github.com/stretchr/testify/assert"
+	"os"
+	"strconv"
+	"testing"
+)
+
+const (
+	envValueString = "envValue"
+	envValueInt    = 123456789
+
+	defaultHostValue = "defaultHost"
+	defaultPortValue = 987654321
+	defaultTypeValue = "defaultType"
+)
+
+func initializeTest(t *testing.T) RegistryInfo {
+	os.Clearenv()
+	return RegistryInfo{
+		Host: defaultHostValue,
+		Port: defaultPortValue,
+		Type: defaultTypeValue,
+	}
+}
+
+func initializeTestWithEnvVariable(t *testing.T, envKey string, envValue string) RegistryInfo {
+	registryInfo := initializeTest(t)
+	if err := os.Setenv(envKey, envValue); err != nil {
+		t.Fail()
+	}
+	return registryInfo
+}
+
+func TestHostEnvVariableUpdatesRegistryInfo(t *testing.T) {
+	registryInfo := initializeTestWithEnvVariable(t, envKeyHost, envValueString)
+
+	registryInfo = OverrideFromEnvironment(registryInfo)
+
+	assert.Equal(t, registryInfo.Host, envValueString)
+}
+
+func TestPortEnvVariableUpdatesRegistryInfo(t *testing.T) {
+	registryInfo := initializeTestWithEnvVariable(t, envKeyPort, strconv.Itoa(envValueInt))
+
+	registryInfo = OverrideFromEnvironment(registryInfo)
+
+	assert.Equal(t, registryInfo.Port, envValueInt)
+}
+
+func TestTypeEnvVariableUpdatesRegistryInfo(t *testing.T) {
+	registryInfo := initializeTestWithEnvVariable(t, envKeyType, envValueString)
+
+	registryInfo = OverrideFromEnvironment(registryInfo)
+
+	assert.Equal(t, registryInfo.Type, envValueString)
+}
+
+func TestHostNoEnvVariableDoesNotUpdateRegistryInfo(t *testing.T) {
+	registryInfo := initializeTest(t)
+
+	registryInfo = OverrideFromEnvironment(registryInfo)
+
+	assert.Equal(t, registryInfo.Host, defaultHostValue)
+}
+
+func TestNoPortEnvVariableDoesNotUpdateRegistryInfo(t *testing.T) {
+	registryInfo := initializeTest(t)
+
+	registryInfo = OverrideFromEnvironment(registryInfo)
+
+	assert.Equal(t, registryInfo.Port, defaultPortValue)
+}
+
+func TestNoTypeEnvVariableDoesNotUpdateRegistryInfo(t *testing.T) {
+	registryInfo := initializeTest(t)
+
+	registryInfo = OverrideFromEnvironment(registryInfo)
+
+	assert.Equal(t, registryInfo.Type, defaultTypeValue)
+}
```

Comment:
@michaelestrin Question -- Have you looked at table driven tests at all in Go? It looks like these methods could utilize that technique, and you could thereby collapse these methods into one with a table containing a series of testable criteria and mocks. I think it's a little more concise. Example: https://github.com/edgexfoundry/edgex-go/blob/2cd11714e75ea178d29e77b27c6dfbe4b8618a15/internal/core/metadata/operators/device/add_test.go#L28

Reply:
Recognize they're idiomatic.  I shy away from them because they tend to be harder for me to read/grok than the old school "one method, one test" approach.  Refactored per your implied request.

Reply:
Yeah, I had that same cognitive clash. Look at them enough and you get used to it though.